src/index.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  FunctionComponent,
+  Fragment,
+  ReactElement,
+  createElement,
+  useState,
+  useEffect,
+  useCallback,
+} from 'react';
+import unified, { PluggableList } from 'unified';
+import remarkParse, { RemarkParseOptions } from 'remark-parse';
+// @ts-ignore
+import remarkToRehype from 'remark-rehype';
+// @ts-ignore
+import rehypeReact from 'rehype-react';
+
+export interface UseRemarkOptions {
+  remarkParseOptions?: Partial<RemarkParseOptions>;
+  // TODO: Pending https://github.com/remarkjs/remark-rehype/pull/13
+  remarkToRehypeOptions?: {};
+  // TODO: Pending
+  rehypeReactOptions?: {};
+  remarkPlugins?: PluggableList;
+  rehypePlugins?: PluggableList;
+  onError?: (err: Error) => void;
+}
+
+export const useRemark = ({
+  remarkParseOptions,
+  remarkToRehypeOptions,
+  rehypeReactOptions,
+  remarkPlugins = [],
+  rehypePlugins = [],
+  onError = () => {},
```

Comment:
This API is different.
It may be possible to create an adapter which is more like the current API on top of this component to smooth the transition.